benchmark-service/benchmark/benchmark.py

```diff
@@ -215,27 +217,59 @@ async def compare(request, userdata):  # pylint: disable=unused-argument
     return await render_template('benchmark', request, userdata, 'compare.html', context)
 
 
-async def query_github(app):
-    global START_POINT
+async def update_commits(app):
+    global benchmark_data
     github_client = app['github_client']
-    request_string = f'/repos/hail-is/hail/commits?since={START_POINT}'
+    batch_client = app['batch_client']
+    gs_reader = app['gs_reader']
 
-    data = await github_client.getitem(request_string)
+    request_string = f'/repos/hail-is/hail/commits?since={START_POINT}'
+    gh_data = await github_client.getitem(request_string)
     new_commits = []
-    for commit in data:
-        sha = commit.get('sha')
-        new_commits.append(commit)
-        log.info(f'commit {sha}')
-        START_POINT = commit['commit']['author'].get('date')
-        log.info(f'start point is now {START_POINT}')
+    formatted_new_commits = []
+    for gh_commit in gh_data:
+
+        sha = gh_commit.get('sha')
+
+        batches = [b async for b in batch_client.list_batches(q=f'sha={sha} running')]
+        try:
+            batch = batches[-1]
+            batch_status = await batch.status()
+        except Exception:  # pylint: disable=broad-except
+            batch_status = None
+
+        file_path = f'gs://{BENCHMARK_TEST_BUCKET_NAME}/benchmark-test/{sha}'
+        has_results_file = gs_reader.file_exists(file_path)
+
+        if not batches and not has_results_file:
+            new_commits.append(gh_commit)
+
     log.info('got new commits')
+    for gh_commit in new_commits:
+        sha = gh_commit.get('sha')
+        batch_id = await submit_batch(sha, batch_client)
```

Comment:
Can you flip the order of batch_client and sha here? Our services code usually has it this way.